static/js/subs.js

```diff
@@ -1287,7 +1287,16 @@ $(function () {
 
     $("#subscriptions_table").on("click", ".sub_unsub_button", function (e) {
         var sub = get_sub_for_target(e.target);
+        var stream_row = $(this).parent();
+
         exports.sub_or_unsub(sub);
+        show_stream_row(this, e);
+        exports.change_state.prevent_once();
+        var stream_id = stream_row.attr("data-stream-id");
+
+        window.location.hash = "#subscriptions" + "/" +
```

Comment:
We have a couple callers to `show_stream_row` that are setting up this hash, and it may make sense to de-duplicate that logic somehow.

Also, this precedes your change, but the `e` argument to to `show_stream_row` is very strange, because we only use it as a boolean.  It would be nice if you did a cleanup commit here first to make the existing codepath a bit cleaner.